### ISSUE TITLE
Strange behaviour if nothing is configured in depguard

### ISSUE THREAD

**Issue by Keyl0ve** [golangci/golangci-lint#5103]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [X] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [X] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/welcome/faq/#why-do-you-have-typecheck-errors).
- [X] Yes, I've tried with the standalone [linter](https://golangci-lint.run/usage/linters/) if available (e.g., gocritic, go vet, etc.).
- [X] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### Description of the problem

If I set depguard as an enable linter and leave depguard's linters-settings unset, I get the following error: Since I have not set any linters-settings, I should expect nothing to be displayed as a result.

```
model/account.go:6:2: import 'github.com/gocraft/dbr/v2' is not allowed from list 'Main' (depguard)
        "github.com/gocraft/dbr/v2"
```

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version v1.60.3 built with go1.23.1 from (unknown, modified: ?, mod sum: "h1:l38A5de24ZeDlcFF+EB7m3W5joPD99/hS5SIHJPyZa0=") on (unknown)
```

</details>

### Configuration

<details>

```console
run:
  concurrency: 2
  modules-download-mode: readonly
  issues-exit-code: 0

linters-settings:
  govet:
    check-shadowing: false

output:
  format: colored-line-number

linters:
  enable:
    - bodyclose
    - depguard
    - goconst
    - gocritic
    - gofmt
    - goimports
    - gosec
    - misspell
    - nakedret
    - prealloc
    - unconvert
    - unparam

issues:
  exclude-rules:
    - path: _test\.go
      linters:
        - errcheck
        - unparam
        - staticcheck

```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.23.2 darwin/arm64
GO111MODULE=''
GOARCH='arm64'
GOBIN=''
GOCACHE='/Users/kyo.sato/Library/Caches/go-build'
GOENV='/Users/kyo.sato/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='arm64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMODCACHE='/Users/kyo.sato/go/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='darwin'
GOPATH='/Users/kyo.sato/go'
GOPRIVATE=''
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/opt/homebrew/Cellar/go/1.23.2/libexec'
GOSUMDB='sum.golang.org'
GOTMPDIR=''
GOTOOLCHAIN='local'
GOTOOLDIR='/opt/homebrew/Cellar/go/1.23.2/libexec/pkg/tool/darwin_arm64'
GOVCS=''
GOVERSION='go1.23.2'
GODEBUG=''
GOTELEMETRY='local'
GOTELEMETRYDIR='/Users/kyo.sato/Library/Application Support/go/telemetry'
GCCGO='gccgo'
GOARM64='v8.0'
AR='ar'
CC='cc'
CXX='c++'
CGO_ENABLED='1'
GOMOD='/Users/kyo.sato/go/src/github.com/everytv/delish-signage-server/go.mod'
GOWORK=''
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'
GOGCCFLAGS='-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -ffile-prefix-map=/var/folders/w7/9xw1bf6s0459ppvy_mjpjxt00000gp/T/go-build1484051970=/tmp/go-build -gno-record-gcc-switches -fno-common'
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v

...

storevision/response/opportunity.go:5:2: import 'github.com/everytv/delish-signage-server/util/time' is not allowed from list 'Main' (depguard)
        utilTime "github.com/everytv/delish-signage-server/util/time"
        ^
storevision/response/opportunity.go:6:2: import 'github.com/gocraft/dbr/v2' is not allowed from list 'Main' (depguard)
        "github.com/gocraft/dbr/v2"
        ^
storevision/response/pagination.go:3:8: import 'github.com/everytv/delish-signage-server/util/paginate' is not allowed from list 'Main' (depguard)
import util "github.com/everytv/delish-signage-server/util/paginate"

...

```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// add your code here
```

</details>

### Validation

- [X] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by alexandear** [comment]
Hey. It's expected behaviour because you are using [`depguard`](https://golangci-lint.run/usage/linters/#depguard) without any configuration. You have to either add a configuration or disable `depguard`. The default rule of `depguard` is to allow only stdlib in all files.

This was answered in [Slack](https://gophers.slack.com/archives/CS0TBRKPC/p1686125047175859):

<img width="693" alt="image" src="https://github.com/user-attachments/assets/f0ebf78e-e27a-480d-954a-115b747c2f6f">
